### PR TITLE
GHA CI improvements

### PR DESCRIPTION
**Pull request by kolyshkin**
A bit of CI facelifting. Please see individual commits for details.

### CODE REVIEW

**Comment by kolyshkin** [comment]
Related: https://github.com/godbus/dbus/pull/291

**Comment by kolyshkin** [comment]
@jsouthworth @guelfey PTAL

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -11,29 +11,28 @@ jobs:
...
-      run: |
-        go get -v -t -d ./...
-        if [ -f Gopkg.toml ]; then
-            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-            dep ensure
-        fi
 
     - name: Build
-      run: go build -v ./...
+      run: go build "${{ matrix.race }}" -v ./...
```
Comment (guelfey) [inline .github/workflows/go.yml:35]: I think you need to omit the quotes here

Comment (kolyshkin) [reply .github/workflows/go.yml:35]: right

**Comment by guelfey** [comment]
Thanks!

**Pull request merged**